pkg/controller/awselb/aws_elb_controller.go

```diff
@@ -265,11 +279,24 @@ func (c *Controller) syncMachine(key string) error {
 
 	region := coMachineSetSpec.ClusterHardware.AWS.Region
 	mLog.Debugf("Obtaining AWS clients for region %q", region)
-	client, err := clustopaws.NewClient(c.kubeClient, coMachineSetSpec, machine.Namespace, region)
+	client, err := c.clientBuilder(c.kubeClient, coMachineSetSpec, machine.Namespace, region)
+	if err != nil {
+		return err
+	}
+
+	status, err := controller.AWSMachineProviderStatusFromClusterAPIMachine(machine)
 	if err != nil {
 		return err
 	}
 
+	if status.LastELBSync != nil {
```

Comment:
Is there anything else about the Machine that we should look at to determine whether an ELB sync may be necessary? For example, if the spec of the Machine changes, then I would expect the controller to sync the ELB even if the resync duration has not elapsed.

Reply:
The only things I can think of are (1) an admin editing a label to tickle the CO into re-adding it, and (2) the instance ID changes in the status indicating a new vm was picked up. As far as the spec goes the only thing I can think of is a machine getting the master role added after the fact, but I'm not sure we'd even want to start down that path.

I could base it on generation though as we do elsewhere, but that wouldn't detect status changes like instance ID.